src/Node/ModuleNode.php

```diff
@@ -41,7 +41,7 @@ public function __construct(Node $body, ?AbstractExpression $parent, Node $block
         if (!$embeddedTemplates instanceof Node) {
             trigger_deprecation('twig/twig', '3.21', \sprintf('Not passing a "%s" instance as the "embedded_templates" argument of the "%s" constructor is deprecated.', Node::class, static::class));
 
-            $embeddedTemplates = new Nodes($embeddedTemplates);
+            $embeddedTemplates = new Nodes($embeddedTemplates ?? []);
```

Comment:
What about leveraging `EmptyNode` if `$embeddedTemplates` is `null` (see #4608)?

Reply:
Thank you for the PR!